pkg/didcomm/protocol/route/service.go

```diff
@@ -384,6 +389,74 @@ func (s *Service) Register(connectionID string) error {
 	return s.saveRouterConnectionID(connectionID)
 }
 
+// AddKey adds a recKey of the agent to the registered router. This method blocks until a response is
+// received from the router or it times out.
+// TODO https://github.com/hyperledger/aries-framework-go/issues/1076 Support for multiple routers
+func (s *Service) AddKey(recKey string) error {
+	// check if router is already registered
+	routerConnID, err := s.getRouterConnectionID()
+	if err != nil && !errors.Is(err, storage.ErrDataNotFound) {
+		return fmt.Errorf("fetch router connection id : %w", err)
+	}
+
+	if routerConnID == "" {
+		return errors.New("router not registered")
+	}
+
+	// get the connection record for the ID to fetch DID information
+	conn, err := s.getConnection(routerConnID)
+	if err != nil {
+		return err
+	}
+
+	// generate message ID
+	msgID := uuid.New().String()
+
+	// register chan for callback processing
+	keyUpdateCh := make(chan KeylistUpdateResponse)
+	s.setKeyUpdateResponseCh(msgID, keyUpdateCh)
+
+	keyUpdate := &KeylistUpdate{
+		ID:   msgID,
+		Type: KeylistUpdateMsgType,
+		Updates: []Update{
+			{
+				RecipientKey: recKey,
+				Action:       add,
+			},
+		},
+	}
+
+	if err := s.outbound.SendToDID(keyUpdate, conn.MyDID, conn.TheirDID); err != nil {
+		return fmt.Errorf("send route request: %w", err)
+	}
+
+	select {
+	case keyUpdateResp := <-keyUpdateCh:
+		if err := processKeylistUpdateResp(recKey, keyUpdateResp); err != nil {
+			return err
+		}
+	// TODO configure this timeout at decorator level
```

Comment:
Any Issue # for this TODO?

Reply:
[Added](https://github.com/hyperledger/aries-framework-go/compare/7a3023b54d29f6ab445c5e2ae619c297a1b2113b..6c12881229ad4bf9c2bc6378ebca772663963458#diff-7cc3df019152fe2c557826a312cc0a12R439).